src/package.json

```diff
@@ -76,7 +75,7 @@
     "@angular/compiler": ">=12.0.0 <14.0.0",
     "@angular/core": ">=12.0.0 <14.0.0",
     "@angular/forms": ">=12.0.0 <14.0.0",
-    "chart.js": ">2.9.0 <3.0.0",
-    "ng2-charts": ">=2.3.0 <3.0.0"
+    "chart.js": ">3.0.0 <4.0.0",
+    "ng2-charts": ">3.0.0 <4.0.0"
```

Comment:
As it stands this is a breaking change which means this is unsuitable to be merged until UX Aspects v6 (likely May time to align when Angular 14).

Reply:
Is there any way to continue supporting 2.9?

Reply:
I don't think this would be possible, v3 has a lot of breaking changes. In their migration guide they mentioned it was necessary to break backwards compatibility to achieve this. https://www.chartjs.org/docs/latest/getting-started/v3-migration.html

Although that said, I can't foresee many other changes being introduced to charts in the mean time. So these changes could sit and wait till Angular 14?

Reply:
Possibly, we would need to remove any reliance on types that are not part of both versions, there aren't many so in some cases, e.g. `Color` we could create our own type that has the same structure.

Also anywhere that we have had to make changes in the plugin we would need to abstract it away, we could do this with a set of utility functions (or potentially an adapter class with two implementations, one for each version), for example, something like this:

```ts
function registerPlugin(instance?: unknown): void {
    // if pluginService exists then we are in v2
    if (window.Chart.pluginService) {
       (window as any).Chart.pluginService.register(new TimelineChartPlugin());
    else {
            instance.register(new TimelineChartPlugin());
    }
}
```

But how possible this is for every case I'm not sure

Reply:
Yes, typescript makes it difficult but I would think that we could probe for capabilities. I would suggest evaluating this approach. Judging from the amount of time invested in this ticket, I anticipate some resistance in moving teams over to v3 even if we do introduce it in a major version increment. Since it's a hard dependency, it will force a team to upgrade their entire application to ng2-charts 3 even if they don't have any need to do so.

Reply:
We may even be able to just have two different versions of the plugin the old one, and the one with the v3 changes, and the appropriate one gets registered based on detecting the version. We would have to strip out a lot of the types, but for this one file I think that shouldn't be a problem

Reply:
Thanks you both for the discussion on this. I shall make an attempt at this, there isn't a massive amount of changes made to the plugin so it may not a massive issue.

Apologies if this is obvious but in terms of the documentation examples, would it make sense to revert them back to v2, it would make reviewing these changes a lot simpler.
From my understanding the plugin should support v2 and v3 but the documentation site would have the v2 installed and it would be up to users to update their charts to v3 if they wish to use it?

Reply:
No, you can keep the v3 examples, it's only example code so changing it doesn't affect anyone who has already used them. The only part which really matters is the plugin.

Reply:
No worries, apologies if I added more work to this than was necessary.